modules/core/shared/src/main/scala/gem/Observation.scala

```diff
@@ -27,8 +26,19 @@ final case class Observation[+T, +S, +D](
 
 object Observation {
 
-  /** A fully specified observation. */
-  type Full = Observation[TargetEnvironment, StaticConfig, Step[DynamicConfig]]
+  /** A fully specified observation, with unknown instrument. */
+  type Full = Full.Aux[i.type] forSome { val i: Instrument }
```

Comment:
If you have an `Observation.Full` will the compiler be able to prevent you from, say, making the static config and steps disagree about the instrument?  If so, is all of the additional complexity buying us much?

Reply:
Yes. That's what this type means.